File: src/main/java/dragonbook/inter/expr/Id.java
package dragonbook.inter.expr;

import dragonbook.lexer.Word;
import dragonbook.symbols.Type;

public class Id extends Expr {
    public final int offset; // relative address

    public Id(Word id, Type type, int offset) {
        super(id, type);
        this.offset = offset;
    }
}

File: src/main/kotlin/dragonbook/inter/expr/Id.kt
package dragonbook.inter.expr

import dragonbook.lexer.Token
import dragonbook.lexer.Word
import dragonbook.symbols.Type

data class Id(
    val id: Word,
    override val type: Type,
    val offset: Int // relative address
) : Expr {

    override val op: Token get() = id
    override fun toString(): String = op.toString()
}